extensions/sql-migration/src/api/utils.ts

```diff
@@ -703,6 +705,205 @@ export async function getManagedInstancesDropdownValues(managedInstances: azureR
 	return managedInstancesValues;
 }
 
+// retrieve the auth keys
+export async function retrieveAuthKeys(migrationStateModel: MigrationStateModel): Promise<SqlMigrationServiceAuthenticationKeys> {
+
+	let defaultKeys: SqlMigrationServiceAuthenticationKeys = {
+		authKey1: '',
+		authKey2: ''
+	};
+
+	const service = migrationStateModel._sqlMigrationService;
+	if (service) {
+		const account = migrationStateModel._azureAccount;
+		const subscription = migrationStateModel._sqlMigrationServiceSubscription;
+		const resourceGroup = migrationStateModel._sqlMigrationServiceResourceGroup.name;
+		const location = service.location;
+		const serviceName = service.name;
+		if (service?.properties?.integrationRuntimeState) {
+			service.properties.integrationRuntimeState = undefined;
+		}
+		const keys = await getSqlMigrationServiceAuthKeys(
+			account,
+			subscription,
+			resourceGroup,
+			location,
+			serviceName);
+
+		return keys;
+	}
+	return defaultKeys;
+}
+
+// invoke and execute the script
+export async function invokeScript(scriptPath: string): Promise<void> {
+
+	var spawn = require("child_process").spawn, child;
```

Comment:
You are using require importing the child_process module. In modern TypeScript generally we use import instead. import allows for better type checking and IntelliSense support.